crates/net/network/tests/it/connect.rs

```diff
@@ -80,3 +86,50 @@ async fn test_connect_with_boot_nodes() {
         dbg!(ev);
     }
 }
+
+#[tokio::test(flavor = "multi_thread")]
+async fn test_incoming_node_id_blacklist() {
+    reth_tracing::init_tracing();
+    let secret_key = SecretKey::new(&mut rand::thread_rng());
+
+    let reth_socket = SocketAddr::new([127, 0, 0, 1].into(), 30305);
+
+    // instantiate geth and add ourselves as a peer
+    let temp_dir = tempfile::tempdir().unwrap().into_path();
+    let geth = Geth::new().data_dir(temp_dir).disable_discovery().spawn();
```

Comment:
@Rjected should we add a helper fn which instantiates Geth with a random datadir?

Reply:
Yeah we should, we will probably need to do it for every geth instance in a test.